### PR TITLE
Used d3 to add ticks to graph on index, rotated axis ticks with d3, removed css

### PR DESCRIPTION
**Pull request by kmcostel**

### CODE REVIEW

File: public/bitemp_graph.js [hunk 1]
```diff
@@ -77,37 +94,57 @@ var barChart = function() {
...
+      
       if (yMax) {
-        maxdate = yMax;
+        maxEnd = yMax;
       }
       else {
-        maxdate =
+        maxEnd =
           moment.max(data.map(function(d){
+            if (!d.content.valEnd.includes('9999')) {
```
Comment (fxue) [inline public/bitemp_graph.js]: what about startsWith(), or indexOf()?

File: public/bitemp_graph.js [hunk 2]
```diff
@@ -77,37 +94,57 @@ var barChart = function() {
...
       }
       else {
-        mindate =
+        minStart =
           moment.min(data.map(function(d){
             return moment(d.content.valStart);
           })).toDate();
       }
+      
+      var MIN_MOMENT = moment('0000-00-00T00:00:00');
```
Comment (fxue) [inline public/bitemp_graph.js]: is this valid date? I prefer 1600-01-01T00:00:00

Comment (kmcostel) [inline public/bitemp_graph.js]: Yes it seems to work fine, but I'll change to yours

File: public/bitemp_graph_helper.js [hunk 3]
```diff
@@ -144,8 +142,88 @@ function edit(uri) {
...
+}
+
+var deleteDoc = function (chart) {
+  var uri = chart.getLogicalURI();
+  var ajax = true;
+  if (!uri) {
+    return;
+  }
+  var collection = getDocColl(uri);
+  var url = '/v1/documents?uri=' + uri + '&temporal-collection='+collection.collections[0];
```
Comment (fxue) [inline public/bitemp_graph_helper.js]: there is no guarantee that temporal collection will always be the first one in the response, check this call out to find all temporal collections:
http://pubs.marklogic.com:8011/REST/GET/manage/v2/databases/[id-or-name]/temporal/collections

File: public/bitemp_graph_helper.js [hunk 4]
```diff
@@ -144,8 +142,88 @@ function edit(uri) {
...
+  var currDate = new Date();
+  
+  $.ajax(
+  {
+    url: 'http://localhost:3000/v1/resources/temporal-range?rs:collection='+collection.collections[0],
```
Comment (fxue) [inline public/bitemp_graph_helper.js]: same with the last comment.

File: public/bitemp_graph_helper.js [hunk 5]
```diff
@@ -144,8 +142,88 @@ function edit(uri) {
...
+      if (tempDate.valueOf() > currDate.valueOf()) {
+        ajax = false;
+        document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
+      }
+    }
+    else if (tempDate.valueOf() > sysBoxDate.valueOf()) {
+      document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
+      ajax = false;
+    }
+
```
Comment (fxue) [inline public/bitemp_graph_helper.js]: can be simplified to 

var sysBoxDate= document.getElementById('sysStartBox').value;
if (!sysBoxDate) { url+= ..; } else sysBoxDate=currentData;
if (tempDate.valueOf() > sysBoxDate.valueOf()){
document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
}
else { your ajax call}

File: public/bitemp_graph.js (hunk 1 above)
Comment (kmcostel) [inline public/bitemp_graph.js]: Changed to startsWith()

File: public/search.js [hunk 6]
```diff
@@ -97,6 +103,19 @@ function displayAxis(times)
...
+    success: function(data, textStatus) {
+      console.log('got collections: ' + data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log('problem');
+    },
+    async: false,
+  });
+}
+
```
Comment (fxue) [inline public/search.js]: This function is defined but not used in search.js. also it is already defined in another .js file.

File: public/bitemp_graph_helper.js (hunk 5 above)
Comment (kmcostel) [inline public/bitemp_graph_helper.js]: Thanks

File: public/search.js (hunk 6 above)
Comment (kmcostel) [inline public/search.js]: I was trying to use the function in the other js file but I couldn't access it. I was having trouble trying to include the search.js file in the index page, so I need help with this

File: public/bitemp_graph_helper.js (hunk 3 above)
Comment (kmcostel) [inline public/bitemp_graph_helper.js]: After I use this http://pubs.marklogic.com:8011/REST/GET/manage/v2/databases/%5Bid-or-name%5D/temporal/collections and get the temporal collections in the database, then should I still use the original ajax call I was using that got all the collections a document belongs to, then check to see if any of those collections are in the temporal collections list I get?

Comment (kmcostel) [inline public/bitemp_graph_helper.js]: Alright I believe this is working

File: public/bitemp_graph_helper.js (hunk 4 above)
Comment (kmcostel) [inline public/bitemp_graph_helper.js]: Fixed

File: index.html [hunk 7]
```diff
@@ -153,6 +145,7 @@
     </div>
 
     <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
+    <!--<script type="text/javascript" src="public/search.js"></script>-->
```
Comment (dmcassel) [inline index.html]: remove

File: public/bitemp_graph.js [hunk 8]
```diff
@@ -77,37 +95,57 @@ var barChart = function() {
...
+          })).toDate();
       }
 
-      console.log('ymin='+mindate,' ymax='+maxdate);
+      maxStart =
+        moment.max(data.map(function(d){
+          return moment(d.content.valStart);
+        })).toDate();
+
+      if (maxEnd.getFullYear() === 1600) { // All end dates are infinty
```
Comment (dmcassel) [inline public/bitemp_graph.js]: Changing really high dates to really low dates seems illogical to me. Why not set maxStart first, then if maxEnd is year 9999, set it to maxStart + 10y as you do below?

File: public/bitemp_graph_helper.js [hunk 9]
```diff
@@ -144,8 +142,136 @@ function edit(uri) {
...
+          response = collArr[i];
+        }
+      }
+    }
+  }
+  return response;
+}
+
+var deleteDoc = function (chart) {
+//var checkTimeRanges = function (chart) {
```
Comment (dmcassel) [inline public/bitemp_graph_helper.js]: Looks like an old line of code; delete

File: public/bitemp_graph_helper.js [hunk 10]
```diff
@@ -144,8 +142,136 @@ function edit(uri) {
...
+  
+  var tempColl;
+  if (collArr && tempCollArr) {
+    collArr = collArr['collections'];
+    tempColl = findCommonColl(collArr, tempCollArr);
+  }
+  
+  $.ajax( //Gets a temporal collection
+  {
+    url: 'http://localhost:3000/v1/resources/temporal-range?rs:collection='+tempColl,
```
Comment (dmcassel) [inline public/bitemp_graph_helper.js]: You can leave out the 'http://localhost:3000'. By doing so, it's one less place to change if the app is run on a different port. Also, suppose you were accessing the application from a remote server -- localhost wouldn't get where you want to go.

File: public/search.js [hunk 11]
```diff
@@ -51,20 +51,21 @@ $('#dropdown').change(function()
 );
 
 //function to make ajax call to get min and max times
-function ajaxTimesCall(selectedColl)
+var ajaxTimesCall = function(selectedColl)
```
Comment (dmcassel) [inline public/search.js]: Either form should be okay; why the change?

File: search.html [hunk 12]
```diff
@@ -91,13 +91,7 @@
         color: #000000;
         font-weight: normal;
         -webkit-box-shadow: 1px 1px 2px rgba(0,0,0,0.3);
-        -moz-box-shadow: 1px 1px 2px rgba(0,0,0,0.3);
-        box-shadow: 1px 1px 2px rgba(0,0,0,0.3);
```
Comment (dmcassel) [inline search.html:95]: box-shadow is the standard form for this property and ought to be kept here. The -webkit and -moz versions are browser-specific extensions.

File: search.html [hunk 13]
```diff
@@ -91,13 +91,7 @@
...
         -webkit-transition: all 0.15s linear;
-        -moz-transition: all 0.15s linear;
-        -ms-transition: all 0.15s linear;
-        -o-transition: all 0.15s linear;
-        transition: all 0.15s linear;
```
Comment (dmcassel) [inline search.html:100]: As above, let's keep transition and drop the prefixed ones.

File: public/search.js (hunk 11 above)
Comment (kmcostel) [inline public/search.js]: Yea no reason; I was trying to get access the function outside of search.js, but I now know this doesn't help at all and that I need to include the js file in the index.html page

File: search.html (hunk 13 above)
Comment (kmcostel) [inline search.html:100]: Are -moz, -ms, and -o the prefixed ones?

Comment (dmcassel) [inline search.html:100]: Yes

File: public/bitemp_graph.js (hunk 8 above)
Comment (kmcostel) [inline public/bitemp_graph.js]: Yea but I still need a way to tell if all the sysEnd times are infinity. I want to find the max sysEnd time that isn't infinity right?

File: search.html (hunk 13 above)
Comment (kmcostel) [inline search.html:100]: Included

File: search.html (hunk 12 above)
Comment (kmcostel) [inline search.html:95]: Included below
